woocommerce/woocommerce
# Checkout: Collapsible Order Summary in mobile view

### Changes proposed in this Pull Request:

We decided not to implement sticky behaviour of #45285 but this solves:

Implement an order summary (on mobile) that is:

✅ collapsed by default
✅ add a final Order summary before Place order button

See https://github.com/woocommerce/woocommerce/issues/45285#issuecomment-2428080020 if you'd like to see the rationale.

When in mobile size screen you'll see a collapsed by default order summary. The desktop experience should behave as before. Additionally at mobile size a duplicate order summary is rendered near bottom of checkout so that the user has convenient access to see the order summary at top or bottom of screen.

There is also some adjustment and shuffling of layout in the mobile mode. E.g. the duplicate summary at bottom of screen  has a header outside of the summary which is a little different from the one rendered in desktop view that has a header inside the summary border. 

I have not updated font sizes of headers to match the design, maybe this could be considered as a follow up to make it look more consistent with the design.

I also fixed a small visual bug in mobile where the `+ Add apartment, suite, etc.` toggle was not spaced correctly.

Screenshots:

<img width="195" alt="Screenshot 2024-10-24 at 11 27 45 AM" src="https://github.com/user-attachments/assets/1bc22058-0055-46fd-91bf-085b6d7370e9">
<img width="196" alt="Screenshot 2024-10-24 at 11 27 56 AM" src="https://github.com/user-attachments/assets/8fd5ec83-39b4-407f-baa5-90aa991d4d32">




### How to test the changes in this Pull Request:

1. View your checkout page with some sample products added to cart.
2. See that the desktop page still has the summary on the right side and a duplicate order summary should *not* be rendered
3. Now view at a mobile size
4. See that there is a collapsed by default order summary header which includes a summary of the total price.
5. Tap it to open the summary and see the rendered summary of order items.
6. Scroll to bottom of order and see that another non-collapsible order summary is rendered. 





### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Checkout: Add a collapsible order summary for smaller screens.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
